packages/grid/_modules_/grid/hooks/features/core/useGridControlState.ts

```diff
@@ -41,8 +43,14 @@ export function useGridControlState(apiRef: GridApiRef) {
         if (hasSubStateChanged) {
           if (controlState.propOnChange) {
             const newModel = newSubState;
+            const details =
+              props.signature === Signature.XGrid
+                ? {
+                    api: apiRef.current,
+                  }
+                : {};
```

Comment:
Just a small detail.

```suggestion
            const details = props.signature === Signature.XGrid ? { api: apiRef.current } : {};
```

The same in `useApi`.